src/main/java/org/web3j/protocol/core/filters/Filter.java

```diff
@@ -39,7 +40,7 @@ public void run(ScheduledExecutorService scheduledExecutorService, long blockTim
             }
 
             filterId = ethFilter.getFilterId();
-            EthLog ethLogInit = web3j.ethGetFilterLogs(filterId).send();
+            EthLog ethLogInit = this.createFilterRequest(this.filterId).send();
```

Comment:
I think it makes sense for this initial request to take place asynchronously too once we have the filter id. Appreciate the current implementation is down to me, but it would be great if you could update this too. E.g.:

```java
scheduledExecutorService.submit(() -> {
                EthLog ethLog = null;
                try {
                    ethLog = this.createFilterRequest(this.filterId).send();
                } catch (IOException e) {
                    throwException(e);
                }
                
                process(ethLog.getLogs());
            });
```